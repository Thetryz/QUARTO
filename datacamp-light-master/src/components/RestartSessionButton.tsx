import * as React from "react";

import Button from "./Button";
import * as styles from "./Button.module.scss";

import noop from "../helpers/noop";

export interface IRestartSessionButtonProps
  extends React.Props<RestartSessionButton> {
  className?: string;
  visible?: boolean;
  onClickRestart?: (event: React.SyntheticEvent<HTMLButtonElement>) => void;
}

export class RestartSessionButton extends React.Component<
  IRestartSessionButtonProps
> {
  public static defaultProps: Partial<IRestartSessionButtonProps> = {
    visible: true,
    onClickRestart: noop,
  };

  public render() {
    if (!this.props.visible) {
      return false;
    }

    return (
      <div className={this.props.className}>
        <Button
          size="small"
          appearance="inverted"
          onClick={this.props.onClickRestart}
        >
          <span>Restart</span>
          <span className={`${styles.largeScreen}`}> Session</span>
        </Button>
      </div>
    );
  }
}

export default RestartSessionButton;
