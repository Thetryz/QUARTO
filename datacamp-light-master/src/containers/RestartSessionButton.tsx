import { connect, MapDispatchToProps, MapStateToProps } from "react-redux";

import { restartSession } from "../redux/backend-session";
import { State } from "../redux";

import {
  RestartSessionButton,
  IRestartSessionButtonProps,
} from "../components/RestartSessionButton";

export interface IOwnProps {
  visible?: boolean;
  className?: string;
}

const mapStateToProps: MapStateToProps<
  IRestartSessionButtonProps,
  IOwnProps
> = (state: State, ownProps) => ({
  visible: ownProps.visible,
  className: ownProps.className,
});

const mapDispatchToProps: MapDispatchToProps<
  IRestartSessionButtonProps,
  IOwnProps
> = dispatch => ({
  onClickRestart: event => dispatch(restartSession()),
});

export default connect(mapStateToProps, mapDispatchToProps)(
  RestartSessionButton
);
