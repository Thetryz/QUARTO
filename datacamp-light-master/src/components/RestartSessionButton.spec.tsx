import * as React from "react";
import { shallow } from "enzyme";

import { RestartSessionButton } from "./RestartSessionButton";

describe("RestartSessionButton", () => {
  it("should exist", () => {
    expect(RestartSessionButton).toBeDefined();
  });

  it("should match snapshot", () => {
    const tree = shallow(<RestartSessionButton />);
    expect(tree).toMatchSnapshot();
  });
});
