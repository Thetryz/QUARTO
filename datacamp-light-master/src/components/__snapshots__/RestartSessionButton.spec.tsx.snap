// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestartSessionButton should match snapshot 1`] = `
<div>
  <Button
    appearance="inverted"
    disabled={false}
    onClick={[Function]}
    size="small"
  >
    <span>
      Restart
    </span>
    <span
      className="largeScreen"
    >
       Session
    </span>
  </Button>
</div>
`;
